### ISSUE TITLE
API Access using basic authentication. 

### ISSUE THREAD

**Issue by finalbeta** [Vidlec/Zabbix.NET#1]
Hello, 

I did not read anything about this and I was wondering if and how we could use the .NET API library when HTTP authentication is enabled in Zabbix (Basic auth)

Some more information about this topic: https://www.zabbix.com/forum/showthread.php?t=39103

**Comment by finalbeta** [comment]
I've forked this and added basic auth to the code. https://github.com/finalbeta/Zabbix.NET
The code is compatible with the original, I've added another constructor that takes a booleon to perform basic auth. 
Only logging out doesn't work for me with the changes I made. It seems to be returning an expandoObject with nothing in it.

**Comment by finalbeta** [comment]
Final update from me. 
The logout method did not work for me. I saw that you where sending "null" as parameter to log out. However, my zabbix installation did not exept that. Zabbix requires an empty params arry according to https://www.zabbix.com/documentation/3.2/manual/api/reference/user/logout
I've changed this inside my fork. 

So now, at least for me, basic auth is working and the logout method works. 
Feel free to pull those things inside your branch, as it should stay compatible with your code.
